### PR TITLE
Remove JMX support

### PR DESCRIPTION
**Pull request by ppkarwasz**
As partial solution to #1344, this PR drop JMX support and replaces the usage of `RingBufferAdmin` in tests with a direct access to the `RingBuffer`.

## Motivation

JMX is an old monitoring technology, which less and less used today. Given its inherent insecurity, we should evaluate if JMX support should even be part of Log4j Core.

Our JMX support code is the main obstacle to the separation of `AsyncLogger` and related code into its own artifact.

IMHO we should drop JMX support in 3.0.x and evaluate if such a technology should be added again in 3.1.x.

### CODE REVIEW

File: log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java [hunk 1]
```diff
@@ -436,13 +435,7 @@ public boolean stop(final long timeout, final TimeUnit timeUnit) {
...
 
             this.setStopping();
-            String name = getName();
-            try {
-                Server.unregisterLoggerContext(name); // LOG4J2-406, LOG4J2-500
-            } catch (final LinkageError | Exception e) {
-                // LOG4J2-1506 Hello Android, GAE
-                LOGGER.error("Unable to unregister MBeans", e);
-            }
+
```
Comment (vy) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java:438]: Why did we remove this and another below Android-specific guards?

**Review by vy — APPROVED** [review]

File: log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java (hunk 1 above)
Comment (ppkarwasz) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java:438]: I am not sure I understand: I removed the `Server` class and method call, so the `try ... catch` becomes empty.

Comment (vy) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java:438]: Oh! :facepalm: Got it.

**Review by vy — APPROVED** [review]

**Comment by ppkarwasz** [comment]
Part of  #2163

**Pull request merged**